src/commands/Admin/roleall.ts

```diff
@@ -0,0 +1,80 @@
+/*
+	Adapted from https://github.com/NotEnoughUpdates/bush-bot/blob/master/src/commands/admin/roleAll.ts
+*/
+import { FireMessage } from "@fire/lib/extensions/message";
+import { Command } from "@fire/lib/util/command";
+import 
+{ GuildMember, 
+  Role,
+  Permissions }  from 'discord.js';
+
+export default class RoleAllCommand extends Command {
+	public constructor() {
+		super('roleAll', {
+			aliases: ['role-all', 'rall'],
+			category: 'admin',
+            userPermissions: [Permissions.FLAGS.ADMINISTRATOR],
+			description: "Give a role to al the members of this server",
+			args: [
+				{
+					id: 'role',
+					type: 'role',
+					prompt: {
+						start: 'What role would you like to give to all the members of this server?',
+						retry: '{error} Pick a valid role.'
+					}
+				}
+			],
+            enableSlashCommand: true,
+            restrictTo: "guild",
+		});
+	}
+
+	async exec(
+        message: FireMessage , 
+        args: { role: Role }) {
+		if (!message.guild) return await message.util.reply('This command can only be run in a server.');
+
+		if (!message.member!.permissions.has('ADMINISTRATOR') )
+			return await message.util.reply(`You must have admin permissions to use this command.`);
+
+		if (args.role.comparePositionTo(message.guild.me!.roles.highest) >= 0 && !args.role) {
```

Comment:
????????

you are calling a method on the rule and also checking !role which will never be true and makes no sense at all.

it should also check that the role is not managed, not higher than the author's role (unless they're the owner) and is not the `@everyone` role.

I have code to do all of this and an error for this already in the language file

Reply:
It works for me but you’re right this isn’t the most efficient way to do this.